cmake/Platform/Linux/PAL_linux.cmake

```diff
@@ -47,3 +47,21 @@ ly_set(LY_PYTHON_CMD ${CMAKE_CURRENT_SOURCE_DIR}/python/python.sh)
 # Note: Only ("xcb" or "wayland" should be considered)
 set(PAL_TRAIT_LINUX_WINDOW_MANAGER "xcb" CACHE STRING "Sets the Window Manager type to use when configuring Linux")  
 set_property(CACHE PAL_TRAIT_LINUX_WINDOW_MANAGER PROPERTY STRINGS xcb wayland)
+
+# Use system default OpenSSL library instead of maintaining an O3DE version for Linux
+include(${CMAKE_CURRENT_LIST_DIR}/OpenSSL_linux.cmake)
+
+cmake_path(RELATIVE_PATH CMAKE_CURRENT_LIST_DIR BASE_DIRECTORY ${LY_ROOT_FOLDER} OUTPUT_VARIABLE openssl_install_code)
+
+set(openssl_install_code "
+# Use system default OpenSSL library instead of maintaining an O3DE version for Linux
+include(${openssl_cmake_rel_directory}/OpenSSL_linux.cmake)
+")
```

Comment:
So there is a tricky issue with this.

the `ly_associate_package` command adds any 3rd party packages name to an [LY_PACKAGE_NAMES global property](https://github.com/o3de/o3de/blob/development/cmake/3rdPartyPackages.cmake#L642).

The creation of the SDK layout using the install command requires that LY_PACKAGE_NAMES global property to replicate the `ly_associate_package` calls into a [generated BuiltinPackages_linux.cmake](https://github.com/o3de/o3de/blob/development/cmake/Platform/Common/Install_common.cmake#L587-L591) file.

The reason the ly_associate_package calls are generated, is due to them being able to be distributed across all CMake files within the O3DE project, including within gems.

Now here is where the issue occurs.
If the machine only where the SDK layout is built on only has OpenSSL 1.1 available, then only the "AWSNativeSDK OpenSSL-1" package gets added to the SDK layout.

If the user of that SDK layout is on a distro that uses OpenSSL 3.0, then the wrong OpenSSL version will be used.

This same issue occurs when building an SDK layout on a machine with OpenSSL 3.0 available.
That SDK layout will find the System OpenSSL, which if it is 1.1, would cause the AWSNativeSDK to fail to find OpenSSL again.

The version of OpenSSL that was found in the cmake configuration for the source engine needs to be "baked" into the openssl_install_code to only look for that specific version.

Reply:
FYI, this code was just 'moved' from the original location which was BuiltinPackages.cmake. @junbo75 was the author who did the original change to use the system OpenSSL

Reply:
Yes, that is a tricky issue. I'd prefer to defer that issue to Prism/installer to come up with a solution for that scenario

Reply:
Ok, that makes more sense now. Created [GHI](https://github.com/o3de/o3de/issues/11640)